src/styles/button-split.scss

```diff
@@ -1,155 +1,164 @@
+        border-radius: var(--fdButton_Border_Radius_Left_RTL);
       }
     }
 
-    &:active {
-      .#{$block}__text {
-        color: var(--sapButton_Selected_TextColor);
+    &:nth-of-type(2) {
+      border-radius: var(--fdButton_Border_Radius_Right);
+
+      @include fd-rtl() {
+        border-radius: var(--fdButton_Border_Radius_Right_RTL);
       }
     }
   }
 
   &--emphasized {
-    > [class*="--emphasized"] {
-      & > .#{$block}__text {
-        color: var(--sapButton_Emphasized_TextColor);
-        font-weight: bold;
-      }
+    @include fd-button-pad(var(--sapButton_Emphasized_Background));
 
-      &:hover {
-        & > .#{$block}__text {
-          color: var(--sapButton_Emphasized_Hover_TextColor);
-        }
+    > .#{$block-button}--emphasized {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Emphasized_TextColor));
+
+      &:first-of-type {
+        @include fd-set-margin-right(var(--fdButton_Menu_Emphasized_Margin));
       }
 
-      &:active {
-        & > .#{$block}__text {
-          color: var(--sapButton_Selected_TextColor);
+      &:nth-of-type(2) {
+        @include fd-rtl() {
+          border-left-width: var(--fdButton_Emphasized_Border_Width);
         }
       }
     }
   }
 
   &--transparent {
-    > [class*="--transparent"] {
-      & > .#{$block}__text {
-        color: var(--sapButton_TextColor);
-      }
+    @include fd-button-pad(var(--sapButton_Lite_Background));
 
-      &:hover {
-        & > .#{$block}__text {
-          color: var(--sapButton_TextColor);
-        }
-      }
+    > .#{$block-button}--transparent {
+      @include fd-split-button-border(var(--fdButton_Menu_Transparent_Border_Width));
+      @include fd-split-button-separator(var(--fdButton_Menu_Transparent_Separator_Display), var(--sapButton_Lite_TextColor));
+    }
+  }
 
-      &:active {
-        & > .#{$block}__text {
-          color: var(--sapButton_Selected_TextColor);
-        }
-      }
+  &--ghost {
+    @include fd-button-pad(var(--fdButton_Ghost_Background));
+
+    > .#{$block-button}--ghost {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_TextColor));
+    }
+  }
+
+  &--positive {
+    @include fd-button-pad(var(--sapButton_Accept_Background));
+
+    > .#{$block-button}--positive {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Accept_TextColor));
+    }
+  }
+
+  &--negative {
+    @include fd-button-pad(var(--sapButton_Reject_Background));
+
+    > .#{$block-button}--negative {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Reject_TextColor));
+    }
+  }
+
+  &--attention {
+    @include fd-button-pad(var(--sapButton_Attention_Background));
+
+    > .#{$block-button}--attention {
+      @include fd-split-button-separator(var(--fdButton_Menu_Separator_Display), var(--sapButton_Attention_TextColor));
```

Comment:
Lots of repetitive code that can be avoided. 
For example, you can move all the specific values like this(for all types of buttons with modifier classes:
```
$button-states: (
  "ghost": (
    "btnPad": var(--fdButton_Ghost_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_TextColor)
  ),
  "positive": (
    "color": var(--sapButton_Accept_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_Accept_TextColor)
  ),
  "negative": (
    "color": var(--sapButton_Reject_Background), 
    "btnSeparatorDisplay": var(--fdButton_Menu_Separator_Display), 
    "btnSeparatorBackground": var(--sapButton_Reject_TextColor)
  ),
  ..... other types
);

```

and then loop through them:
```
@each $set-name, $props-set in $button-states {
  &--#{$set-name} {
    color: map_get($props-set, "color");

    @include fd-button-pad(map_get($props-set, "color"));

    > .#{$block-button}--negative {
      @include fd-split-button-separator(map_get($props-set, "btnSeparatorDisplay"), map_get($props-set, "btnSeparatorBackground"));
    }
  }
}
```